cmd/thanos/compact.go

```diff
@@ -110,6 +110,10 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 	compactionConcurrency := cmd.Flag("compact.concurrency", "Number of goroutines to use when compacting groups.").
 		Default("1").Int()
 
+	includeShards := cmd.Flag("shard.include", "Prometheus shard to compact. May be specified multiple times. Cannot be used with --shards.exclude.").Strings()
```

Comment:
I think something like this is great, but I have 2 suggestions:
* I would not tie it to Prometheus. There can be many sources of blocks, including Thanos-receiver. This being said it should be agnosting to Prometheus. It should be agnostic to `sharding` as well as again, it unnecessarly ties things. I would literally match of `external-labels of source blocks`. As we do literally that on compactor. It's tied to TSDB blocks. So e.g repeated `block-external-label-matcher` flag would be nice?
* I would not reinvent the wheel in terms how to specify this "string". Especially this form does not specify what exactly we include / exclude on. Pod name? replica name? external labels? UIDs? (: I would stick to matchers because that is what we do, we query the blocks like metrics (where each block is single series). We can even reuse internal code for `github.com/prometheus/tsdb@v0.8.0/labels/selector.go` `labels.Matcher`. So `matcher` flag could take literally `{cluster=~"monzo-super-cluster-(east|west)", zone="abc"}`. What do you think?

In this manner we don't need include/exclude as well (:

Reply:
This sounds good - I'm happy not to tie it to prometheus shard and reuse matcher code 👍